### ISSUE TITLE
How Auto-WCAG related to ACT

### ISSUE THREAD

**Issue by WilcoFiers** [act-rules/act-rules.github.io#42]
Todo

**Comment by josehilera** [comment]
I have commpared ACT with Auto-WCAG rules template, and these are the results.

|ACT | Auto WCAG
|----|---------
|Section 3.1: "Each ACT Rule MUST provide descriptive title" | OK, field "name". 
|Section 3.1: "Each ACT Rule MUST provide unique identifier" | OK, field "rule_id".
|Section 3.2: "Each ACT Rule MUST have a description" | OK, field "Description". 
|Section 3.3: "Each ACT Rule MUST identify the accessibility requirements to which the rule maps, for example, WCAG 2.0 success criterion" | OK, field "success_criterion". 
|Section 3.4: "Each ACT Rule MUST provide a list of limitations" | REVIEW, because no specific field exists for it. It exists the field "Background", confirm if it is to describe limitations.
|Section 3.4: "Each ACT Rule MUST provide a list of assumptions" | OK, field "Assumptions"
|Section 3.4: "Each ACT Rule MUST provide a list of exceptions" | REVIEW, because no specific field exists for it. It exists the field "Background", confirm if it is suitable to describe limitations.
|Section 4: "The following test subject types are common in accessibility testing: HTTP Response, DOM Tree, Rendered Page, Template, Script" | REVIEW the accepted values for the field "environment", because the names are not the same as ACT, and they do not include types as Template or Script. The allowed values are: Markup Document / DOM Structure / Web Browser / WebDriver 
|Section 4: "Other types of test subjects may be possible. In those cases the ACT Rule MUST include a detailed description of the test subject type" | REVIEW, because the "WebDriver" type is not include in ACT list, and must be described.
|Section 5.1: Test procedure include a "Selector" | OK, field "selector".
|Section 5.2: "Each rule contains one or more test cases" | REVIEW, because until now, Auto-WCAG rules have only one test case. Maybe, could be a requirement for Auto-WCAG that every rule has only one test case, this is compatible with ACT. But there are rules with a lot of steps, as SC1-1-1-text-alternative.md, with 18 steps. Preciselly, ACT offers this related exmaple: "A rule for HTML img elements has two test cases: 1.Check if there is a text alternative, 2.Check if the image is marked as decorative." In the description of rule SC1-1-1-text-alternative.md apperars: "This test checks that non-text elements conveying information are providing sufficient textual alternatives or, if purely decorative, are hidden from assistive technologies". So its seemd as candidate to split y two different test cases for the same rule.
|Section 5.2: Each test case MUST provide mode (whether test case steps are automated, semi-automated or require manual testing), steps, result, description of the error causing any failing result | REVIEW, because now the mode is assigned to the complete rule with the general field "test_mode", but not to each test case. It is not a problem if it is a requirement that every Auto-WCAG rules contain only a test case.   
|Section 6: "To ensure rules are kept small and atomic, they SHOULD be put into a rule-group instead" | REVIEW, until now there is not Auto-WCAG rule groups, but there are rules too much long, maybe they are candidates to split into small and stomic rules. For example SC1-1-1-text-alternative.md, with 18 steps.
|Section 7: "ACT Rules SHOULD include a warning, if there are significant accessibility support concerns known about a rule" | REVIEW, No accessibility support section in Auto-WCAG rule template.
|Section 8: Output data MUST have at least the following properties: test (Rule ID), subject (Web page), pointer (Selected item), outcome (Passed, Failed or Undetermined) | REVIEW, because in the rule template is not included a field "pointer", although it is used in rules exmaples. Also, "Undetermined" outcome in ACT is named as "CannotTell" in Auto-WCAG.
|Section 9.1: An implementation test consists of two parts: a piece of input data and an expected result. | REVIEW, No sub-sections in implementation test section available in Auto-WCAG rule template.

**Comment by WilcoFiers** [comment]
@josehilera That's a great analyses. I'm keeping a close eye on how ACT relates to auto-WCAG (I chair both groups, so that's pretty easy). So far most of what ACT has done is based on Auto-WCAG work. ACT has received some input that may mean diverging from the current format a little more. We'll have to see how that works out. I think we may want to keep this analyses around and keep it updated as the drafts progress.

**Comment by annethyme** [comment]
@WilcoFiers, can we close this one as it seems pretty outdated?
